src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -911,6 +912,13 @@ else if((forceReplace || newRecipe.getMustExist()) && duplicateFound != recipeTo
      */
     public String getJobDisplayName()
     {
+        if(jobDisplayName.isEmpty())
+        {
+            //This will only happen if getJobDisplayName is called before the first citizen is assigned. 
+            ICitizenData citizen = colony.getCitizenManager().createAndRegisterCivilianData();
+            jobDisplayName = this.createJob(citizen).getName();
+            colony.getCitizenManager().removeCivilian(citizen);
```

Comment:
Might this here cause issues? @someaddons

Reply:
hm not sure tbh, though this is a pretty weird solution to just get a name, seems like a hacky solution with a high overhead. Should be able to solve this much more smoothly

Reply:
I just checked and it seems to be not used at all? so could just remove it

Reply:
This *is* used, in both PublicWorkerCraftingProductionResolver and PublicWorkerCraftingRequestResolver. In both of those cases, it shouldn't go through this codepath. The codepath that *might* hit his is the one in WindowTownHall.java, and then only once. 

This is the result of building.getJobName returning a non-localization key, and being needed by the crafting code. This is the only contact the building has today with the actual job, and to get the jobName localization key.

Reply:
Couldn't we return a "not loaded" or so then?

Reply:
This is serversided and the stuff you're talking about is clientside

Reply:
We could, but then that would show to the user, and they won't know how to fix it (assign a worker to the building). This is specifically the list of job names in the town hall. Right now, the town hall will show 0/1 for a blank job name if the building has never been assigned a worker before this change.

Reply:
This code is serverside, and is serialized to the window for display clientside.

Reply:
where is this used in serialization? Unless intelliJ is lying to me its a function that never gets called

Reply:
Great point, I just looked at the serialization again, and it's referencing the field. It should be calling this function. Fixing.